src/Networking/MQTT/mqtt_pal.cpp

```diff
@@ -53,4 +54,27 @@ ssize_t mqtt_pal_recvall(mqtt_pal_socket_handle fd, void* buf, size_t sz, int fl
 	}
 
 	return res;
+}
+
+void mqtt_pal_mutex_init(mqtt_pal_mutex_t *mutex)
+{
+	static uint8_t count = 0;
+	constexpr char format[] = "MQTTClient%d";
+	// Size should be fine, can accomodate up to 99; while we don't expect
+	// more than a couple.
+	char name[sizeof(format)] = { 0 };
+	SafeSnprintf(name, sizeof(name), format, count);
+	mutex->m = new Mutex();
+	mutex->m->Create(name);
```

Comment:
This is not safe because the mutex will point to a name in memory that has been overwritten. As the names of mutexes are only used for debugging purposes and then only rarely, we could consider giving them all the same name.

Reply:
I had mistakenly thought that `Create` makes a copy of the string, modified to use the same name. As you said, it's only for debugging.